src/lib/DataConnectStore.js

```diff
@@ -139,9 +139,20 @@ export default class DataConnectStore {
   }
 
   updateAccount (connectorId, accountIdx, values) {
-    let connector = this.find(c => c.id === connectorId)
-    connector.accounts[accountIdx] = values
-    return this.putConnector(connector)
+    let connector = this.findConnectorById(connectorId)
+    let previousAccount = Object.assign({}, connector.accounts[accountIdx])
+
+    connector.accounts[accountIdx].auth.password = values.password
```

Comment:
I thinks this method `updateAccount` should update not only the password but also all data from the form here. And we will be able to update any data of an account by this way.